document/core/exec/instructions.rst

```diff
@@ -182,6 +182,489 @@ Where the underlying operators are non-deterministic, because they may return on
+
+6. Push the value :math:`\V128.\CONST~c` to the stack.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~(\V128\K{.}\CONST~c_2)~(\V128\K{.}\CONST~c_3)~\V128\K{.}\vsternop &\stepto& (\V128\K{.}\CONST~c)
+     & (\iff c = \vsternop_{\I128}(c_1, c_2, c_3)) \\
+   \end{array}
+
+
+.. todo::
+   shuffle, load splat, load extend.
+
+
+.. _exec-simd-splat:
+
+:math:`\shape\K{.}\SPLAT`
+.........................
+
+1. Let :math:`t` be the type :math:`\unpacked(\shape)`.
+
+2. Assert: due to :ref:`validation <valid-simd-splat>`, a value of :ref:`value type <syntax-valtype>` :math:`t` is on the top of the stack.
+
+3. Pop the value :math:`t.\CONST~c_1` from the stack.
+
+4. Let :math:`n` be the integer :math:`\lanes(\shape)`.
+
+5. Let :math:`c` be the result of :math:`\simdof_{\shape}(c_1^n)`.
+
+6. Push the value :math:`\V128.\CONST~c` to the stack.
+
+.. math::
+   \begin{array}{lcl@{\qquad}l}
+   (t\K{.}\CONST~c_1)~\shape\K{.}\SPLAT &\stepto& (\V128\K{.}\CONST~c)
+     & (\iff t = \unpacked(\shape)
+       \wedge c = \simdof_{\shape}(c_1^{\lanes(\shape)}))
+     \\
+   \end{array}
+
+
+.. _exec-simd-extract-lane:
+.. _exec-simd-extract-lane-sx:
+
+:math:`\shape\K{.}\EXTRACTLANE\K{\_}\sx^?~\laneidx`
+...................................................
+
+1. Assert: due to :ref:`validation <valid-simd-extract-lane>`, :math:`\laneidx < \lanes(\shape)`.
+
+2. Assert: due to :ref:`validation <valid-simd-extract-lane>`, a value of :ref:`value type <syntax-valtype>` |V128| is on the top of the stack.
+
+3. Pop the value :math:`\V128.\CONST~c_1` from the stack.
+
+4. Let :math:`n` be the integer :math:`\lanes(\shape)`.
+
+5. Let :math:`i_1^n` be the sequence :math:`\simdto_{\shape}(c_1)`.
+
+6. Let :math:`t_1` be the type :math:`\packed(\shape)`.
+
+7. Let :math:`t_2` be the type :math:`\unpacked(\shape)`.
+
+8. Let :math:`c_2` be the result of computing :math:`\extend^{sx^?}_{t_1,t_2}(i_1^n[\laneidx])`.
+
+9. Push the value :math:`t_2.\CONST~c_2` to the stack.
+
+.. math::
+   \begin{array}{l}
+   \begin{array}{lcl@{\qquad}l}
+   (\V128\K{.}\CONST~c_1)~\V128\K{.}\EXTRACTLANE~\laneidx &\stepto& (t_2\K{.}\CONST~c_2)
+   \end{array}
+   \\ \qquad
+     \begin{array}[t]{@{}r@{~}l@{}}
+      (\iff & n = \lanes(\shape) \\
+       \wedge & t_1 = \packed(\shape) \\
+       \wedge & t_2 = \unpacked(\shape) \\
+       \wedge & c_2 = \extend^{sx^?}_{t_1,t_2}(\simdto_{\shape}(c_1)[\laneidx])
+     \end{array}
+   \end{array}
+
+
+.. _exec-simd-replace-lane:
+
+:math:`\shape\K{.}\REPLACELANE~\laneidx`
+........................................
+
+1. Assert: due to :ref:`validation <valid-simd-replace-lane>`, :math:`\laneidx < \lanes(\shape)`.
+
+2. Let :math:`t_1` be the type :math:`\unpacked(\shape)`.
+
+3. Assert: due to :ref:`validation <valid-simd-replace-lane>`, a value of :ref:`value type <syntax-valtype>` :math:`t_1` is on the top of the stack.
+
+4. Pop the value :math:`t_1.\CONST~v_1` from the stack.
```

Comment:
why `v_1` instead of `c_1`?

Reply:
I wanted to differentiate them a little more, since `c_1` is always a `v128`, but `v_1` is `i32` or `i64`. I'll change it to `c`, it's more consistent.